### PR TITLE
Use custom user agent string for aiohttp

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/coresys.py [hunk 1]
```diff
@@ -88,6 +89,11 @@ def __init__(self):
         self._jobs: Optional[JobManager] = None
         self._security: Optional[Security] = None
 
+        # Set default header for aiohttp
+        self._websession._default_headers = MappingProxyType(
+            {aiohttp.hdrs.USER_AGENT: SERVER_SOFTWARE}
```
Comment (pvizeli) [inline supervisor/coresys.py:94]: I think that can be difficult for ingress? Maybe we just set `ClientSession(..., headers={USER_AGENT})` or do you think it's not an issue if we overwrite that?

Comment (ludeeus) [reply supervisor/coresys.py:94]: It should not be an issue (we already do this on the core side), but I can test ingress later today

**Review by pvizeli — APPROVED** [review]
If it's not an issue with ingress

File: supervisor/coresys.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/coresys.py:94]: Tested ingress with FileEditor, VSCode, Node-RED, and my 2 test add-ons, no issues with any of them :+1:

**Pull request merged**